### DISCUSSION TITLE
[Errno 8] nodename nor servname provided, or not known

### DISCUSSION THREAD

**Discussion by 101Dude** [mborsetti/webchanges#69] [Q&A]
I need help figuring this out. It's been happening for a while now.

My jobs run without error and then three or four times a day I get the error  `[Errno 8] nodename nor servname provided, or not known`

When this error happens many jobs are listed in my email reporter, but not all of them (~7 out of 11 total). Whereas when they run successfully I am getting a single job reporting a change successfully (I would think if there was a connection problem all of my jobs would fail with the same error).

I test each one using webchanges --test-filter [job number] and they run successfully.

I don't have a clue where to start troubleshooting this issue.

Thank you

```
Software:
• webchanges: 3.19
• CPython: 3.12.0 ('main', 'Nov 21 2023 17:38:04') Clang 15.0.0 (clang-1500.0.40.1)
• SQLite: 3.43.2

System:
• Platform: macOS-14.3.1-x86_64-i386-64bit, x86_64
• Processor: i386
• CPUs (logical): 16
• Free memory: 83.8G physical plus 0.0B swap.
• Free disk '/': 164.1G (94.6%)

Installed PyPi dependencies:
• beautifulsoup4: 4.12.2
• chump: 1.6.0
• cssbeautifier: 1.14.11
  - jsbeautifier: 1.14.11
  - six: 1.16.0
  - editorconfig: 0.12.3
• cssselect: 1.2.0
• deepdiff: 6.7.1
  - ordered-set: 4.1.0
  - pyyaml: 6.0.1
• h2: 4.1.0
  - hyperframe: 6.0.1
  - hpack: 4.0.0
• html2text: 2020.1.16
• html5lib: 1.1
  - six: 1.16.0
  - webencodings: 0.5.1
  - lxml: 4.9.3
  - lxml: 4.9.3
• httpx: 0.25.2
  - anyio: 4.2.0
  - certifi: 2023.7.22
  - idna: 3.6
  - sniffio: 1.3.0
• jq: 1.6.0
• jsbeautifier: 1.14.11
  - six: 1.16.0
  - editorconfig: 0.12.3
• keyring: 24.3.0
  - jaraco.classes: 3.3.0
• lxml: 4.9.3
  - cssselect: 1.2.0
  - html5lib: 1.1
  - BeautifulSoup4: 4.12.2
• markdown2: 2.4.12
• msgpack: 1.0.7
• pdftotext: 2.2.2
• Pillow: 10.1.0
  - markdown2: 2.4.12
  - packaging: 23.2
• platformdirs: 4.1.0
• playwright: 1.40.0
  - greenlet: 3.0.1
  - pyee: 11.0.1
  - typing-extensions: 4.9.0
• psutil: 5.9.7
• pypdf: 4.0.2
  - typing_extensions: 4.9.0
• pytesseract: 0.3.10
  - packaging: 23.2
  - Pillow: 10.1.0
• pyyaml: 6.0.1
• redis: 5.0.1
  - typing-extensions: 4.9.0
  - requests: 2.31.0
• requests: 2.31.0
  - charset-normalizer: 3.3.2
  - idna: 3.6
  - urllib3: 2.1.0
  - certifi: 2023.7.22
  - PySocks: 1.7.1
• vobject: 0.9.6.1
  - python-dateutil: 2.8.2

Playwright browser:
• Name: chromium
• Version: 121.0.6167.139
• Free memory with browser loaded: 83.8G physical plus 0.0B swap
```

**Comment by mborsetti** [comment]
Hello,

Errors starting with brackets come from the Operating System, so hard to reproduce -- especially for me, as I don't run on MacOS.  Have you tried Googling the error and see if you get any inspiration?  

Does it happen on any specific kind of jobs (i.e. only those without ``use_browser: true``)? Does specifying a low number of ``--max-workers`` to reduce parallelism help?

Please do share any findings so they can be added to the documentation.

**Reply by 101Dude** [reply]
Thanks for the tips. I ran a search and found a lead... will see if that helps and report back.

**Reply by 101Dude** [reply]
It looks like it could be something in python code? Would you please take a look and see if this is the issue. 

https://www.delftstack.com/howto/python/socket.gaierror-errno-8-nodename-nor-servname-provided-or-not-known/

**Reply by mborsetti** [reply]
Sadly that site is junk: ``python2`` (version 2) has been superseded by ``python3`` since 2008! Also, ``sockets`` is not called in this code (but may be called by an underlying library).

Can you do a run with ``-vv`` and, if you get that error, upload the log?
